addons/pos_event/models/pos_order.py

```diff
@@ -0,0 +1,69 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    attendee_count = fields.Integer('Attendee Count', compute='_compute_attendee_count')
+
+    @api.depends('lines.event_registration_ids')
+    def _compute_attendee_count(self):
+        for order in self:
+            order.attendee_count = len(order.lines.mapped('event_registration_ids'))
+
+    def action_view_attendee_list(self):
+        action = self.env["ir.actions.actions"]._for_xml_id("event.event_registration_action_tree")
+        action['domain'] = [('pos_order_id', 'in', self.ids)]
+        return action
+
+    @api.model
+    def sync_from_ui(self, orders):
+        results = super().sync_from_ui(orders)
+        paid_orders = self.browse([order['id'] for order in results['pos.order'] if order['state'] in ['paid', 'done', 'invoiced']])
+
+        if not paid_orders:
+            return results
+
+        line_with_event = paid_orders.mapped('lines').filtered(lambda line: line.event_ticket_id)
+
+        # Verify seats availability before creating registrations
+        for line in line_with_event:
+            ticket_seat_available = line.event_ticket_id.seats_available
+            event_seat_available = line.event_ticket_id.event_id.seats_available
+            if ticket_seat_available < line.qty and ticket_seat_available != 0 or event_seat_available < line.qty and event_seat_available != 0:
+                raise UserError(_('No more available seats for %s') % line.event_ticket_id.name)
+
+        # Create registrations
+        for line in line_with_event:
+            for _count in range(int(line.qty)):
```

Comment:
```suggestion
            for _ in range(int(line.qty)):
```

Reply:
Cannot do that it shadow the l10n method